lib/modules/ai_bot/site_settings_extension.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+module DiscourseAi::AiBot::SiteSettingsExtension
+  module ClassMethods
+    def ai_bot_enabled=(val)
+      super(val)
+      DiscourseAi::AiBot::SiteSettingsExtension.enable_or_disable_ai_bots
+    end
+
+    def ai_bot_enabled_chat_bots=(val)
+      super(val)
+      DiscourseAi::AiBot::SiteSettingsExtension.enable_or_disable_ai_bots
+    end
+  end
+
+  def self.enable_or_disable_ai_bots
+    enabled_bots = SiteSetting.ai_bot_enabled_chat_bots.split("|")
+    enabled_bots = [] if !SiteSetting.ai_bot_enabled
+    DiscourseAi::AiBot::EntryPoint::BOTS.each do |id, bot_name, name|
+      active = enabled_bots.include?(name)
+      user = User.find_by(id: id)
+
+      if active
+        if !user
+          user =
+            User.new(
+              id: id,
+              email: "no_email_#{name}",
+              name: bot_name.titleize,
+              username: UserNameSuggester.suggest(bot_name),
+              active: true,
+              approved: true,
+              admin: true,
+              moderator: true,
+              trust_level: TrustLevel[4],
+            )
+          user.save!(validate: false)
+        else
+          user.update!(active: true)
+        end
+      elsif !active && user
+        # will include deleted
+        has_posts = DB.query_single("SELECT 1 FROM posts WHERE user_id = #{id} LIMIT 1").present?
```

Comment:
This complicates things a bit, but does the AI bot also own things like uploads, or post chat messages? Can it own topics?

Reply:
Might be safer to always deactivate?

Reply:
very unlikely, I think it is practically impossible for it just to own stuff like uploads, that said longer term maybe if we add a chat modality it will own chat messages and no posts

I prefer to use the hammer here cause it lets us delete a few thousand accounts on our hosting

Reply:
Yep no problem, just a thought for future 👍